sql/cell_value.cc

```diff
@@ -1015,6 +1053,115 @@ CellValue::printInterval() const
+                {
+                    Json::Value v;
+                    v["num"] = std::signbit(floatVal) ? "-NaN" : "NaN";
+                    context.writeJson(v);
+                }
+            else if (std::isinf(floatVal))
+                {
+                    Json::Value v;
+                    v["num"] = std::signbit(floatVal) ? "-Inf" : "Inf";
+                    context.writeJson(v);
+                }
+            else
+                {
+                    context.writeDouble(toDouble());
+                }
+                
+            return;
+        }           
+    case CellValue::UTF8_STRING:
+    case CellValue::ASCII_STRING:
+        switch (type) {
+        case ST_UTF8_SHORT_STRING:
+            context.writeStringUtf8((const char *)shortString, (size_t)strLength);
+            return;
+        case ST_ASCII_SHORT_STRING:
+            context.writeString((const char *)shortString, (size_t)strLength);
+            return;
+        case ST_UTF8_LONG_STRING:
+            context.writeStringUtf8(longString->repr, (size_t)strLength);
+            return;
+        case ST_ASCII_LONG_STRING:
+            context.writeString(longString->repr, (size_t)strLength);
+            return;
+        default:
+            throw HttpReturnException(400, "unknown string cell type");
+            return;
+        }
+    case CellValue::TIMESTAMP: {
+        context.startObject();
+        context.startMember("ts");
+        context.writeString(toString());
+        context.endObject();
+        return;    
+    }
+    case CellValue::TIMEINTERVAL: {
+        context.startObject();
+        context.startMember("interval");
+        context.writeString(toString());
+        context.endObject();
+        return;         
+    }
+    case CellValue::BLOB: {
+        context.startObject();
+        context.startMember("blob");
+        context.startArray();
+            
+        // Chunks of ASCII are written as a string; non-ASCII is
+        // as integers.
+        const unsigned char * p = blobData();
+        const unsigned char * e = p + blobLength();
+
+        while (p < e) {
+            const unsigned char * s = p;
+            while (p < e && *p >= ' ' && *p < 127 && isascii(*p))
+                ++p;
+            size_t len = p - s;
+            //cerr << "len = " << len << endl;
+            if (len == 1) {
+                context.newArrayElement();
+                context.writeInt(*s);
+            }
+            else if (len >= 2) {
+                context.newArrayElement();
+                context.writeString((const char *)s, len);
+            }
+                
+            while (p < e && (*p <= ' ' || *p >= 127 || !isascii(*p))) {
+                context.newArrayElement();
+                context.writeInt(*p++);
+            }
+        }
+        context.endArray();
+        context.endObject();
+        return;
+    }
+    default:
+        throw HttpReturnException(400, "unknown cell type");
```

Comment:
I'm leaving this one.  The code was simply moved.  There are lots of other places in this file that should probably be a 500; we can address them separately.